test/services/company.profile.service.spec.unit.ts

```diff
@@ -0,0 +1,98 @@
+import { CompanyProfile } from "ch-sdk-node/dist/services/company-profile";
+import CompanyProfileService from "ch-sdk-node/dist/services/company-profile/service";
+import Resource from "ch-sdk-node/dist/services/resource";
+import { ObjectionCompanyProfile } from "../../src/model/objection.company.profile";
+import { getCompanyProfile } from "../../src/services/company.profile.service";
+
+jest.mock("ch-sdk-node/dist/services/company-profile/service");
+
+const ACCESS_TOKEN = "KGGGUYUYJHHVK1234";
+
+describe("company profile service unit tests", () => {
+  const mockGetCompanyProfile = (CompanyProfileService.prototype.getCompanyProfile as jest.Mock);
+
+  beforeEach(() => {
+    mockGetCompanyProfile.mockReset();
+  });
+
+  it("converts company number to uppercase", async () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
+    const company = await getCompanyProfile("sc100079", ACCESS_TOKEN);
+    expect(company.incorporationDate).toEqual("26 June 1872");
+    expect(mockGetCompanyProfile).toBeCalledWith("SC100079");
+  });
+
+  it("returns an ObjectionCompanyProfile object", async () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
+    const company = await getCompanyProfile("00006400", ACCESS_TOKEN);
+    expect(company).toEqual(expectedProfile);
+  });
+
+  it("returns a 400 status code when company is not found", async () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(notFoundSDKResponse);
+    try {
+        await getCompanyProfile("", ACCESS_TOKEN);
+    } catch (e) {
+        expect(e.status).toEqual(400);
+    }
+    expect.assertions(1);
+  });
+});
+
+const dummySDKResponse: Resource<CompanyProfile> = {
```

Comment:
Could this go into a separate util file? It looks like something we may reuse.

Reply:
I want to avoid using shared util files for test data as each test should be independent from each other and have it's own test data. Besides, this is the only file that will get company info from the SDK.